### ISSUE TITLE
Does it work on Windows?

### ISSUE THREAD

**Issue by alex-shamshurin** [prettier/vim-prettier#44]
Prettier cli works from windows console, but `:Prettier` does not.

**Comment by mitermayer** [comment]
Hi @alex-shamshurin,

Will start looking into this and do a release to fix it.

Thank you for reporting!

**Comment by alex-shamshurin** [comment]
Sorry, but is there any progress? I think it's only replace slashes...

**Comment by mitermayer** [comment]
@alex-shamshurin I will look into this this weekend and do a release to fix it but please feel free to submit a PR in the meantime.

**Comment by mitermayer** [comment]
I have just downloaded a windows virtual machine and will investigate this issue tomorrow.

**Comment by mitermayer** [comment]
Did not have a chance to look into this yet, if anyone has some time feel free to pick it up. Will try to find some time to look into it tomorrow

**Comment by alex-shamshurin** [comment]
Ok, I think it's very easy, but I do not know vim-script

**Comment by mitermayer** [comment]
I have just tested this on windows and could not replicate the above described bug. 

***

Using vim from windows powershell:

![output](https://user-images.githubusercontent.com/1042131/29549855-663a5184-86bf-11e7-86d0-f90820e62e6b.gif)

***

Using gvim:

![output](https://user-images.githubusercontent.com/1042131/29549939-dfc50e4a-86bf-11e7-8240-4e6e059255f3.gif)

Happy to re-open or create a new issue if having more context around it.

**Comment by alex-shamshurin** [comment]
Hmm, how to debug it? Prettier really works from command line (from project node_modules\.bin\) and :Prettier command does not provide any error output. It just do nothing.

**Comment by mitermayer** [comment]
Hi @alex-shamshurin,

There are 3 ways that `vim-prettier` can find the `prettier` executable

***

**A - It looks to find the `./node_modules` for the binary in the path of the current parent folder**

*** 

**B - If [_A_] fails then -  It looks for a global `prettier` installation**

***

**C - If [_B_]  fails then - It looks for the `prettier` installation that is installed inside `vim-prettier` itself\***

_\* - As per the readme this assumes you have either used `vim-plug` post-install to do it or gone manually to the `vim-prettier` folder and did a `npm install` or `yarn install`_

***

Would be great if you could provide more information on how is it failing for you so that I can better assist you. 

- Are you starting vim from the folder that has `prettier` installed ?
- Do you have prettier installed globally ?
- Have you installed `vim-prettier` with a post-install script or did a manual `prettier` install inside the pluggin folder ?

**Comment by alex-shamshurin** [comment]
1) Yes, prettier is in devDependencies and installed inside my project and it works on my mac and on Linux, but does not work on Windows
2) No
3) I've just run standart :PlugInstall
4) No messages! Clean history, no errors, warning, etc

**Comment by alex-shamshurin** [comment]
Please reopen this issue, until it solved

**Comment by boojinks** [comment]
Have you set any of the configuration variables for vim-prettier?

I ask because I've noticed a separate [issue ](https://github.com/prettier/vim-prettier/issues/46) where the async version does nothing yet reports no errors.

**Comment by mitermayer** [comment]
Hi @alex-shamshurin, 

I have re-opened this issue to try to help solving your problem, I still cannot reproduce this issue that you have described. I have however confirmed [the bug where PrettierAsync is not working on windows](https://github.com/prettier/vim-prettier/issues/46) created by @boojinks .

However the `:Prettier` command works as expected. Can you please provide me your `vim` version ? Could you also try running vim with just `vim-prettier` enabled to see if that could be some incompatibilities with some of your vim settings / pluggins ?

Thanks for your patience, happy to work together with you to make sure you are unblocked.

**Comment by alex-shamshurin** [comment]
My vim version is:
![vim-version](https://user-images.githubusercontent.com/1595209/29657600-ca8bc732-88c0-11e7-99ea-6be01579bddd.jpg)

I've disabled all other plugins except `vim-prettier` and it still does not work and says nothing.
All I new about vim-script it's a `echom` command.  So what I suggest, -  I can insert `echom` to some functions to debug. In what functions should I insert it to find out what happened and why the plugin is silent?

**Comment by alex-shamshurin** [comment]
Also I have disabled all plugin settings and it did not help

**Comment by alex-shamshurin** [comment]
Hmm, I added some debug points to `prettitier.vim` and I discovered that when I run :Prettitier (NOT :PrettitierAsync) it goes to async instead of sync mode

```

    if l:async && v:version >= 800 && exists('*job_start')
      echom "Before Async"
      call s:Prettier_Exec_Async(l:cmd, l:startSelection, l:endSelection)
    else
      echom "Before Sync"
      call s:Prettier_Exec_Sync(l:cmd, l:startSelection, l:endSelection)
    endif

```
result is "Before Async"

```
function! s:Prettier_Exec_Async(cmd, startSelection, endSelection) abort
  echom "In Exec ASync"
  if s:prettier_job_running != 1
      let s:prettier_job_running = 1
      call job_start(a:cmd, {
        \ 'in_io': 'buffer',
        \ 'in_top': a:startSelection,
        \ 'in_bot': a:endSelection,
        \ 'in_name': bufname('%'),
        \ 'err_cb': {channel, msg -> s:Prettier_Job_Error(msg)},
        \ 'close_cb': {channel -> s:Prettier_Job_Close(channel, a:startSelection, a:endSelection)}})
  endif
endfunction

```
message is  "In Exec ASync

if change `l:async=0` it works

```
function! prettier#Prettier(...) abort
  echom "In Prettier"
  let l:execCmd = s:Get_Prettier_Exec()
  echom l:execCmd
  let l:async = a:0 > 0 ? a:1 : 0
  **let l:async = 0**
 ....
```
It gies sync mode and did job.

**Comment by boojinks** [comment]
I suspected it might be the same problem I was having. 

I don't have access to put in a pull request at the moment but the below should fix it for you.

```
function! s:Prettier_Exec_Async(cmd, startSelection, endSelection) abort
  if s:prettier_job_running != 1
      let s:prettier_job_running = 1

      if has('win32') || has('win64') 
          let l:correct_cmd = 'cmd.exe /c ' . a:cmd
      else
          let l:correct_cmd = a:cmd
      endif
      
      call job_start(l:correct_cmd, {
        \ 'in_io': 'buffer',
        \ 'in_top': a:startSelection,
        \ 'in_bot': a:endSelection,
        \ 'in_name': bufname('%'),
        \ 'err_cb': {channel, msg -> s:Prettier_Job_Error(msg)},
        \ 'close_cb': {channel -> s:Prettier_Job_Close(channel, a:startSelection, a:endSelection)}})
  endif
endfunction
```

If it does, could you do me a massive favour and put in a PR?

Thank you.

**Comment by mitermayer** [comment]
Thank you both of you @boojinks  and @alex-shamshurin  for working together on this issue, will test the above code and submit a PR for it.

One thing that I cannot understand is why on @alex-shamshurin  he was going to the async code path, the default `vim-prettier`  async setting is to disabled, unless he has added the `async` setting to his `.vimrc` regardless will submit a fix and do a release for async now.

Once again thank you all for working together on this issue :)

**Comment by alex-shamshurin** [comment]
Yes, I have this option set.

**Comment by alex-shamshurin** [comment]
I disabled that option and :Prettitier works fine now, even without patching the file with `let l:async = 0`. But I was confused because this config works fine on Mac and on Linux. I think on windows it should ignore `g:prettier#exec_cmd_async=1` or somehow need repair async mode as @boojinks suggested.

**Comment by mitermayer** [comment]
@alex-shamshurin I have just done a release for version 0.0.14 with that fix
